### ISSUE TITLE
Dynamic workspaces don't sometimes remove themselves.

### ISSUE THREAD

**Issue by jtagcat** [paperwm/PaperWM#240]
Usually when you leave a workspace empty and switch from it, the workspace gets destroyed.
![image](https://user-images.githubusercontent.com/38327267/74091807-e7fde900-4ab3-11ea-971f-5fccfd545822.png)
When using PaperWM, the behaviour sometimes happens, but often treats workspaces as they were static.
Related #35

**Comment by jtagcat** [comment]
As a starter, being able to manually 'destroy' a workspace would be nice.
For investigation, could I somehow list everything what PaperWM or gnome thinks is on the workspace.

**Comment by olejorgenb** [comment]
This is actually by design - from https://github.com/paperwm/PaperWM/releases/tag/34.1:

> dynamic workspaces: Keep a minimum number of workspaces (by default 4). This brings consistent workspace ordering for a select number of workspaces, with the added benefit of always having a fresh workspace. (use `gsettings set org.gnome.desktop.wm.preferences num-workspaces n` to configure the number of «static» workspaces).

Ie. set the num-workspaces to 1 to get a fully dynamic workspace experience

**Comment by jtagcat** [comment]
"with the added benefit of always having a fresh workspace"

Well if I want a new workspace,
a) when on the last workspace, switch one workspace down (there's always an empty workspace as the last one, dynamically created) 
![image](https://user-images.githubusercontent.com/38327267/74092004-71161f80-4ab6-11ea-92b6-7bf1714cdbc3.png)
b) in overview, drag a window to the new workspace

I guess a keyboard shortcut for 'insert new workspace above/below' and 'move this workspace 1 above/below' would be nice.

**Comment by jtagcat** [comment]
![image](https://user-images.githubusercontent.com/38327267/74092442-df111580-4abb-11ea-97b5-a8be3b703805.png)
so I set the num-workspaces to 1, keyboard shortcuts work fine, but overview still shows an empty workspace between
when I try to switch to it in the overview, it switches to the one above that workspace

**Comment by hedning** [comment]
I'm guessing you have another monitor? We make sure number of workspaces >= number of monitors, as we need a workspace on each monitor.

I'm guessing the empty workspace is on a secondary monitor. After activating the empty workspace, pointer motion on primary will probably reactivate the workspace on primary. Not sure why it activates the first workspace and not the one that was last active on primary though. 

Basically the overview doesn't work great with multiple monitors, as we break a bunch of gnome-shell's assumption by having monitor specific workspaces. Ideally we want something like #192.

**Comment by jtagcat** [comment]
> I'm guessing the empty workspace is on a secondary monitor. After activating the empty workspace, pointer motion on primary will probably reactivate the workspace on primary. Not sure why it activates the first workspace and not the one that was last active on primary though.
what?

also btw this happened with single-monitor as well

**Comment by jtaala** [comment]
Fixed in #577.
